fitbenchmarking/fitting_benchmarking.py

```diff
@@ -168,35 +168,40 @@ def do_fitting_benchmark_one_problem(prob, minimizers, use_errors=True, count=0,
                                                             minimizer=minimizer_name,
                                                             cost_function=cost_function)
             t_end = time.clock()
+
             print("*** with minimizer {0}, Status: {1}, chi2: {2}".format(minimizer_name, status, chi2))
-            print("   params: {0}, errors: {1}".format(params, errors))
-
-            def sum_of_squares(values):
-                return np.sum(np.square(values))
-
-            if fit_wks:
-                sum_err_sq = sum_of_squares(fit_wks.readY(2))
-                # print " output simulated values: {0}".format(fit_wks.readY(1))
-                if sum_err_sq <min_sum_err_sq:
-                    tmp=msapi.ConvertToPointData(fit_wks)
-                    best_fit=data(minimizer_name,tmp.readX(1),tmp.readY(1))
-                    min_sum_err_sq=sum_err_sq
-            else:
-                sum_err_sq = float("inf")
-                print(" WARNING: no output fit workspace")
-            print("   sum sq: {0}".format(sum_err_sq))
+            
+            sum_err_sq = -1
+            if not status == 'failed':
+                print("   params: {0}, errors: {1}".format(params, errors))
+                def sum_of_squares(values):
```

Comment:
can we have this function definition outside of this function? I was a little surprised to find it in here.